### PR TITLE
Introduce lookup options when inserting node in a NodeMap

### LINKED ISSUE #72: Enable lookup in Data.Graph.Inductive.NodeMap
`NodeMap` only allows insertion of nodes with automatic duplicate prevention. It does not support checking whether a node exists or reporting whether mkNode actually created a new node, though this feature should be easy to add.

Some graph construction algorithms involve inserting some initial nodes to an empty graph and operating recursively on newly-inserted nodes, where the recursion generates additional nodes to insert. If a generated node already exists in the graph, the algorithm does not recurse into it, so the algorithm needs to know whether a generated node already exists.

NodeMap doesn't support this use case. It can be supported by a variant of `mkNode` that additionally returns a boolean indicating whether the node already existed or was newly inserted. Another approach is to add a `lookupNode` function, which though more general, may be less performant because then when processing a  new node (not yet inserted into the graph) there will be two lookups instead of one, one lookup from `lookupNode` and one lookup from `mkNode`.

### PR DESCRIPTION
**Pull request by nobrakal**
This is a try to fix #72 .

The main new function is `mkLookupNode`, which is `mkNode` plus a Boolean saying if the node was already present.

I also added a `member`-like function.

### CODE REVIEW

File: Data/Graph/Inductive/NodeMap.hs [hunk 1]
```diff
@@ -107,6 +125,11 @@ insMapNode m a g =
     let (n, m') = mkNode m a
     in (insNode n g, m', n)
 
+insMapLookupNode :: (Ord a, DynGraph g) => NodeMap a -> a -> g a b -> (Bool, g a b, NodeMap a, LNode a)
```
Comment (ivan-m) [inline Data/Graph/Inductive/NodeMap.hs]: Documentation please.  What is the purpose of this function?

**Comment by ivan-m** [comment]
I'm wondering whether it might be worth defining a new type isomorphic to `Bool` to make it clear what the result it (does `True` mean "added in" or "already present"?).

File: Data/Graph/Inductive/NodeMap.hs (hunk 1 above)
Comment (nobrakal) [reply Data/Graph/Inductive/NodeMap.hs]: Oops yes, I forgotten to document that. It is the function asked in the issue.

**Comment by nobrakal** [comment]
It might worth it, you are right. Seems a bit overkill for two functions. I don't know if `lookup` counter-part functions can be made/are interesting in `NodeMap`

I am also wondering if a direct `lookup` version of `insNode` and maybe `insEdge` were interesting in the `Graph` module... They can be easily optimized in `PatriciaTree`

**Comment by runeksvendsen** [comment]
Is there anything preventing this from being pulled? I'm looking for this exact functionality.

**Comment by runeksvendsen** [comment]
The only thing I'd like to add is that -- while we're at it -- it would be nice to also have the corresponding monadic versions of e.g. `lookupNode`:

`lookupNodeM :: (Ord a) => a -> NodeMapM a b g (Maybe Node)`.

**Comment by cdupont** [comment]
I also need this function :)

**Comment by athas** [comment]
Is there still interest in this functionality?  If so, I will merge this PR.

**Comment by Montmorency** [comment]
This is useful for quickly getting nodes from the label set.

**Pull request merged**